src/Common/src/Interop/Unix/libc/Interop.opendir.cs

```diff
@@ -9,6 +9,6 @@ internal static partial class Interop
     internal static partial class libc
     {
         [DllImport(Libraries.Libc, SetLastError = true)]
-        internal static extern SafeDirHandle opendir(string name);
+        internal static extern Microsoft.Win32.SafeHandles.SafeDirHandle opendir(string name);
```

Comment:
Could we move opendir and closedir into the native shim at the same time as readdir?  They all go hand-in-hand, and I think where possible it'd be good to move functions in groups, so that, for example, we're not going directly to libc for the functions that open and close the handle but to System.Native for interpreting that handle.